designs/FloatingStableAndPrerelease.md

```diff
@@ -0,0 +1,95 @@
+
+For example:
+
+| Floating Version | Available Versions | Versions that match |
+| -----------------|--------------------|-----------------------|
+| 5.*-rc.* | 4.9.0 <br> 5.1.0 <br> 5.2.0-beta.1 <br> 5.2.0-rc.3 <br> 6.0.0-rc.1 | 5.1.0 <br> 5.2.0-rc.3 |
+
+Note that 5.2.0-beta.1 and 6.0.0-rc.1 do not match.
+
+The behavior is best illusrated with examples
+
+| Floating Version | Available Versions | Matching Versions | Best version | Notes |
+| -----------------|--------------------|-------------------|--------------|-------|
+| *                | 1.1.0 <br> 1.2.0-rc.1 <br> 1.2.0-rc.2 <br> 2.0.0 <br> 3.0.0-beta.1 | 1.1.0 <br> 2.0.0 | 2.0.0 | Only stable versions are matched | 
+| 1.*              | 1.1.0 <br> 1.2.0-rc.1 <br> 1.2.0-rc.2 <br> 2.0.0 <br> 3.0.0-beta.1 | 1.1.0 | 1.1.0 | Only stable versions starting 1.* are matched | 
+| 1.2.0-*          | 1.1.0 <br> 1.2.0-rc.1 <br> 1.2.0-rc.2 <br> 2.0.0 <br> 3.0.0-beta.1 | 1.2.0-rc.1 <br> 1.2.0-rc.2 | 1.2.0-rc.2 | The highest matching prerelease is selected |
+| *-*              | 1.1.0 <br> 1.2.0-rc.1 <br> 1.2.0-rc.2 <br> 2.0.0 <br> 3.0.0-beta.1 | 1.1.0 <br> 1.2.0-rc.1 <br> 1.2.0-rc.2 <br> 2.0.0 <br> 3.0.0-beta.1 | 3.0.0-beta.1 | 3.0.0 is the highest version |
+| 1.*-*              | 1.1.0 <br> 1.2.0-rc.1 <br> 1.2.0-rc.2 <br> 2.0.0 <br> 3.0.0-beta.1 | 1.1.0 <br> 1.2.0-rc.1 <br> 1.2.0-rc.2 | 1.2.0-rc.2 | A prerelease version is the highest matching version |
+| *-rc.*              | 1.1.0 <br> 1.2.0-rc.1 <br> 1.2.0-rc.2 <br> 2.0.0 <br> 3.0.0-beta.1 | 1.1.0 <br> 1.2.0-rc.1 <br> 1.2.0-rc.2 <br> 2.0.0 | 2.0.0 | A stable version is the highest matching version |
+
+For convenience, the implementation of this proposal has been uploaded to a helper service where you can try out the version parsing and the version selection. https://nugettoolsdev.azurewebsites.net/5.3.0
+
+You can switch the versions to compare the behavior.
+
+## Considerations
+
+*  Why not use an extra parameter like `<PackageReference Include="NuGet.Packaging" Version="1.*" IncludePrerelease="true" />`
+
+There are a few reasons why we decided against that approach:
+
+1. The PackageReference opt in into prerelease version for ranges is dependant on the version requested. PackageReference allows prereleases as potential versions for dependencies due to (link issue)
+1. You can not specify which prerelease versions you want included. For  example, you can not express the following version from the proposed approach `5.1.*-rc.*`. 
+1. It adds an additional element which could realistically be added anywhere. PackageReference is integrated within MSBuild, you could easily add 
+1. It does not apply in certain situations, creating redundancies.
+
+* How will old clients behave
+
+Old clients fail to parse this version. This is arguably the desired behavior, as it allows the customer to recognize very early that things will not work the way they expect them to. 
+

```

Comment:
Could you provide an example error message that an old client will dump out? In general it would be nice if someone searches for that error message they should know that they must update client to use the feature via docs.

Reply:
Unfortunately that's difficult. 
The reason for that is the old clients can't differentiate between legit failures and ones that might work in the future.

Reply:
That's fine, even if it's a generic message it's good to understand what people will be seeing

Reply:
Cool, I'm adding that to the spec.